packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -73,8 +90,8 @@ function initialize(obj, properties) {
     let keyNames = Object.keys(properties);
 
     for (let i = 0; i < keyNames.length; i++) {
-      let keyName = keyNames[i];
-      let value = properties[keyName];
+      var keyName = keyNames[i];
+      var value = properties[keyName];
```

Comment:
🤔 no?

Reply:
Ah the reason for this is to avoid the "would cause a new closure" error. @snewcomer I think you could fix this by hoisting the `let` outside of the loop itself, that way it's clearer where the `let` exists